### ISSUE TITLE
--sanitizers configuration option lacks documentation

### ISSUE THREAD

**Issue by vpax** [zeek/zeek-docs#240]
As best as I can tell, the `--sanitizers` option is not documented anywhere other than (in a quite limited form) some discussion in the Spicy documentation. A basic question is which ones are available under what compilers/platforms.

**Comment by awelzel** [comment]
> A basic question is which ones are available under what compilers/platforms.

I think this we should leave up to other documentation and the developer to understand compiler and platform features. As likely it's stale/version dependent, too. Maybe instead  document in a generic way that `--sanitizers` sets `-fsanitize=...` compiler option and that there's some special handling of `undefined` and optimizations?

We could provide links to fsanitize for GCC and clang if that's helpful? 

https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html

**Comment by timwoj** [comment]
> I think this we should leave up to other documentation and the developer to understand compiler and platform features.

I agree. Saying something to the effect of "passing sanitizer names to --sanitizers will enable them for the build. See your compiler documentation for what sanitizers are available". We should also mention that `configure` validates whether the options passed will build with the compiler in use.

**Comment by vpax** [comment]
> We could provide links to fsanitize for GCC and clang if that's helpful?

Yes, something like that so it's clear where to start hunting. Right now it's fairly mysterious to someone coming to it afresh.
